src/pymor/reductors/neural_network.py

```diff
@@ -294,24 +297,153 @@ def build_basis(self):
+            Dict of additional parameters for the POD-method.
+        ann_mse
+            If `'like_basis'`, the mean squared error of the neural network on
+            the training set should not exceed the error of projecting onto the basis.
+            If `None`, the neural network with smallest validation error is
+            used to build the ROM.
+            If a tolerance is prescribed, the mean squared error of the neural
+            network on the training set should not exceed this threshold.
+            Training is interrupted if a neural network that undercuts the
+            error tolerance is found.
+        """
+
+        def __init__(self, fom, training_set, validation_set=None, validation_ratio=0.1,
+                     basis_size=None, rtol=0., atol=0., l2_err=0., pod_params=None,
+                     ann_mse='like_basis'):
+            assert 0 < validation_ratio < 1 or validation_set
+            self.__auto_init(locals())
+
+
+        def _compute_layers_sizes(self, hidden_layers):
+            """Compute the number of neurons in the layers of the neural network
+            (make sure to increase the input dimension to account for the time)."""
+            return [len(self.fom.parameters) + 1,] + hidden_layers + [len(self.reduced_basis),]
+
+
+        def _build_rom(self):
+            """Construct the reduced order model."""
+            with self.logger.block('Building ROM ...'):
+                rom = NeuralNetworkInstationaryModel(self.fom.T, self.nt, self.neural_network,
+                                                     self.fom.parameters, name=f'{self.fom.name}_reduced')
+
+            return rom
+
+
+        def build_basis(self):
+            """Compute a reduced basis using proper orthogonal decomposition."""
+            with self.logger.block('Building reduced basis ...'):
+
+                # compute snapshots for POD and training of neural networks
+                with self.logger.block('Computing training snapshots ...'):
+                    U = self.fom.solution_space.empty()
+                    for mu in self.training_set:
+                        u = self.fom.solve(mu)
+                        if hasattr(self, 'nt'):
+                            assert self.nt == len(u) - 1
+                        else:
+                            self.nt = len(u) - 1
+                        U.append(u)
+
+                # compute reduced basis via POD
+                reduced_basis, svals = pod(U, modes=self.basis_size, rtol=self.rtol / 2.,
+                                           atol=self.atol / 2., l2_err=self.l2_err / 2.,
+                                           **(self.pod_params or {}))
+
+                self.training_data = []
+                for i, mu in enumerate(self.training_set):
+                    sample = self._compute_sample(mu, U[i*(self.nt+1):(i+1)*(self.nt+1)], reduced_basis)
+                    for elem in sample:
+                        self.training_data.append(elem)
```

Comment:
```suggestion
                    self.training_data.append(sample)
```

Reply:
I think you mean `extend` here as well:

```suggestion
                    self.training_data.extend(sample)
```

Reply:
Sorry, yes.